### PR TITLE
Add improved docs

### PR DESCRIPTION
**Pull request by wooorm**
### Initial checklist

*   [x] I read the support docs 
*   [x] I read the contributing guide 
*   [x] I agree to follow the code of conduct 
*   [x] I searched issues and couldn’t find anything (or linked relevant results below) 
*   [x] If applicable, I’ve added docs and tests

### Description of changes

[View rendered](https://github.com/remarkjs/remark-rehype/blob/docs/readme.md)

Related-to: unifiedjs/unified#168.

/cc @nemo-omen @remcohaszing @ChristianMurphy @Murderlon @with-heart @joostdecock @mgan59

### CODE REVIEW

**Review by wooorm — COMMENTED** [review]
✅

**Comment by wooorm** [comment]
Please take extra care reviewing this one friends, as it’s such a common use case!

File: readme.md [hunk 1]
```diff
@@ -96,42 +176,326 @@ example.md: no issues found
...
+
+## CSS
 
-###### `options`
+Assuming you know how to use (semantic) HTML and CSS, then it should generally
+be straight forward to style the HTML produced by this plugin.
+With CSS, you can get creative and style the results as you please.
 
-Passed to [`mdast-util-to-hast`][to-hast].
+Some semistandard features, notable [`remark-gfm`][remark-gfm]s tasklists and
```
Comment (with-heart) [inline readme.md:453]:
```suggestion
Some semistandard features, notably [`remark-gfm`][remark-gfm]s tasklists and
```

File: readme.md [hunk 2]
```diff
@@ -96,42 +176,326 @@ example.md: no issues found
...
+footnotes, generate HTML that be unintuitive, as it matches exactly what GitHub
+produces for their website.
+There is a project, [`sindresorhus/github-markdown-css`][github-markdown-css],
+that exposes the stylesheet that GitHub uses for rendered markdown, which might
+either be inspirational for more complex features, or be used as-is to exactly
+match how GitHub styles rendered markdown.
```
Comment (with-heart) [inline readme.md:459]:
```suggestion
either be inspirational for more complex features, or can be used as-is to exactly
match how GitHub styles rendered markdown.
```

**Review by with-heart — APPROVED** [review]
Given that the concept behind `remark-rehype` could be fairly complex to a new user, I think you've done a great job explaining the purpose in the "What is this?" and "When should I use this?" sections. Love it!

**Comment by wooorm** [comment]
thanks for the feedback @with-heart!

File: readme.md [hunk 3]
```diff
@@ -8,32 +8,108 @@
...
+*   [License](#license)
+
+## What is this?
+
+This package is a [unified][] ([remark][]) plugin that switches from remark (the
+markdown ecosystem) to rehype (the HTML ecosystem).
+It does this by transforming the current markdown (mdast) syntax tree into an
+HTML (hast) syntax tree.
+remark plugins deal with mdast and rehype plugins deal with hast, so plugins
+used after `remark-rehype` have to be rehype plugins.
```
[the remark this answers is not in the document]
Comment (wooorm) [reply readme.md:41]: I don’t think it’ll help here, I think the links would distract rather than help with an understanding of “What is this?”

**Pull request merged**